server/swap/swap.go

```diff
@@ -1863,9 +1939,14 @@ func (s *Swapper) handleInit(user account.AccountID, msg *msgjson.Message) *msgj
 	s.latencyQ.Wait(&wait.Waiter{
 		Expiration: expireTime,
 		TryFunc: func() wait.TryDirective {
-			return s.processInit(msg, params, stepInfo)
+			res := s.processInit(msg, params, stepInfo)
+			if res == wait.DontTryAgain {
+				stepInfo.actor.status.endSwapSearch() // contract now recorded
+			}
```

Comment:
Aren't you already doing this in `processInit`? I prefer this pattern if there's a choice. Obviously a slight functional difference, but I don't know if using this pattern (waiting to `endSwapSearch` until after acks are sent) introduces any bottlenecks.

Reply:
Oh right.  The code here is how I started, but then I felt weird about waiting to unblock to send the ack and counterparty audit requests.  Lemme look again.

Reply:
I agree this is way cleaner though here.  It's why I was tempted to to add that EndFun to Waiter mentioned here: https://github.com/decred/dcrdex/pull/1563/files#diff-c456beb63edc783dd7172356d814f0e516f0f7ec8a3ba4f3b180ff24464c2afdR40
I started down that path but it got messier for some reason.

I think this PR needs to be run with loadbot a bit.

Reply:
Ugh, well that's why I moved it into `processInit`... the darn tests use the ack as the indication to move on to the next request in the test.  I'll try to figure out how to modify the tests instead of making the swapper code weird